src/wovnio/wovnphp/Headers.php

```diff
@@ -198,16 +198,26 @@ public function computePathLang()
             } else {
                 $request_uri = $this->env['REQUEST_URI'];
             }
-            preg_match($rp, $server_name . $request_uri, $match);
-            if (isset($match['lang'])) {
-                $lang_code = Lang::formatLangCode($match['lang'], $this->store);
-                if (!is_null($lang_code)) {
-                    $this->pathLang = $lang_code;
+
+            $full_url = $server_name . $request_uri;
+            $lang_identifier = null;
+            if ($this->store->settings['url_pattern_name'] == 'custom_domain') {
+                foreach ($this->store->settings['custom_domain_langs'] as $lang_url => $lang) {
+                    if (strpos($full_url, $lang_url) !== false) {
```

Comment:
Is this safe? I know this is an unlikely scenario but what if we have setting `['custom_domain_langs'] = {"test.com"}`, but the url is `https://test.com.something.else.com/index.php`?

Reply:
I agree. Please try to stay as close as you can to the wovn.cs implementation. We should be checking the host component against all of the custom domain languages, followed by the path.

Reply:
I will review wovn.cs and copy as much as possible.